src/main/java/com/csv/communitytrackerjava/service/ProjectServiceImpl.java

```diff
@@ -80,13 +80,39 @@ public ProjectResponseDTO findAllProject() {
         payloadDTO.setAdditionalProperty("projects", projectMapper.toListDTO(projectRepository.findAll()));
         return toProjectResponseDTO("Successfully fetch all projects.", payloadDTO);
     }
-    
-    public ProjectResponseDTO toProjectResponseDTO(String message, ProjectPayloadDTO payloadDTO){
+
+    @Override
+    public ProjectResponseDTO deleteProject(int id) throws Exception {
+        ProjectPayloadDTO payloadDTO = new ProjectPayloadDTO();
+        Project projectFound = projectRepository.findById(id)
```

Comment:
You can transfer this logic to a helper method as it was used by different methods, then you can use the logic checkIfItsInactive helper method to that method as it is being called right after getting the record.